### PR TITLE
feat: melBands as separate feature extractor

### PR DESCRIPTION
**Pull request by jamesb93**
This separates out the melBands extraction into its own extractor as this can be a useful feature to use.

This is draft as tests still need to be written, and I haven't thought of a good set of data to test on.

﻿- feat: separate melBands into its own extractor
- feat: expose melBands in needed places in code

### CODE REVIEW

**Comment by hughrawlinson** [comment]
Cool! Is there something about the existing test audio that makes it unsuitable for testing this feature?

**Comment by hughrawlinson** [comment]
Ah I see - maybe essentia, librosa, or yaafe have an equivalent extractor?

**Comment by jamesb93** [comment]
@hughrawlinson thanks for your patience!

I've now implemented a test based on computing melBands on the VALID_AMPLITUDE_SPECTRUM provided in the test suite. I did this by taking the data and passing it through the FluCoMa `fluid.melbands~` object which returns the same results!

File: __tests__/extractors/melBands.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+describe("melBands", () => {
+  test("should return correct melBands value given a valid signal", (done) => {
+    var en = melBands({
+      sampleRate: 44100,
+      bufferSize: 512,
+      ampSpectrum: TestData.VALID_AMPLITUDE_SPECTRUM,
+      melFilterBank: utilities.createMelFilterBank(26, 44100, 512),
+    });
+
+    console.log(en);
```
Comment (hughrawlinson) [inline __tests__/extractors/melBands.ts:16]:
```suggestion
```

File: src/extractors/extractorUtilities.ts [hunk 2]
```diff
@@ -5,6 +5,5 @@ export function mu(i: number, amplitudeSpect: Float32Array): number {
     numerator += Math.pow(k, i) * Math.abs(amplitudeSpect[k]);
     denominator += amplitudeSpect[k];
   }
```
Comment (hughrawlinson) [inline src/extractors/extractorUtilities.ts:7]:
```suggestion
  }
  
```

**Comment by hughrawlinson** [comment]
This is great, thanks! I added a suggestion to remove a console.log that shouldn't be in there.

**Comment by jamesb93** [comment]
> This is great, thanks! I added a suggestion to remove a console.log that shouldn't be in there.

bwah. Good catch - sorry about that.

**Pull request merged**